### PR TITLE
Rename the `native` gc option to `default`

### PR DESCRIPTION
**Pull request by ghost**
## Summary
Rename the new `native` gc option to `default` for improved clarity.
Made the `native` option internal only.

## Details
I raised a discussion about the naming of the new `native` gc option
due to it potentially being confusing depending on the backend.
We agreed to rename it to `default` to better fit what this option
actually means. I made the `native` option not choosable via
command-line switches as it should be an internal implementation detail.

---

## Notes for Reviewers
I mostly just renamed the changes from this pr https://github.com/nim-works/nimskull/pull/692.

<!--
Pull Request(PR) Help

Before Merge Ensure:
* title reads like a short changelog line entry
* code includes tests and is documented
* leave the source better than before, but split out big reformats

See contributor (guide)[https://nim-works.github.io/nimskull/contributing.html]
for details, especially if you're new to this project.

Tips that make PRs easier:
* for big/impactful changes, start with chat/discussions to refine ideas
* refine the pull request message over time; don't have to nail it in one go
* handle the single commit message requirement at the end of review.

### CODE REVIEW

**Comment by saem** [comment]
I don't think this is quite right, because the GC on the JS and VM targets is `native` to them, so simply replacing `native` with `default` now makes it more confusing for those.

File: compiler/front/optionsprocessor.nim [hunk 1]
```diff
@@ -644,7 +644,7 @@ func testCompileOptionArg*(conf: ConfigRef; switch, arg: string): CompileOptArgC
   case switch.normalize
   of "gc":
     case arg.normalize
-    of "native": asResult conf.selectedGC == gcNative
+    of "default": asResult conf.selectedGC == gcDefault
```
Comment (zerbina) [inline compiler/front/optionsprocessor.nim]:
```suggestion
    of "default":
      let default =
        case conf.backend
        of backendC:             gcOrc
        of backendJs, backendVm: gcNative
        of backendInvalid:       unreachable() 
      asResult conf.selectedGC == default
```

The `testCompilationOptionArg` is used to implement the `compileOption("gc", x)` magic, so a `gcDefault` cannot reach here.

Having said that, I currently can't come up with a situation where querying whether the default GC is selected makes sense, so I'm not sure if doing so should be allowed.

Is there a use-case you have in mind?

File: compiler/front/optionsprocessor.nim [hunk 2]
```diff
@@ -1154,8 +1154,8 @@ proc processSwitch*(switch, arg: string, pass: TCmdLinePass,
     expectArg(switch, arg)
     if pass in {passCmd2, passPP}:
       case arg.normalize
-      of "native":
-        conf.selectedGC = gcNative
+      of "default":
+        conf.selectedGC = gcDefault
```
Comment (zerbina) [inline compiler/front/optionsprocessor.nim]:
```suggestion
        conf.selectedGC = gcUnselected
```

File: compiler/front/main.nim [hunk 3]
```diff
@@ -441,7 +441,7 @@ proc customizeForBackend*(graph: ModuleGraph, conf: ConfigRef,
     of excNone, excNative: conf.exc = excGoto
     of excGoto:            discard
 
-    if conf.selectedGC in {gcUnselected, gcNative}:
+    if conf.selectedGC in {gcUnselected, gcDefault}:
```
Comment (zerbina) [inline compiler/front/main.nim]:
```suggestion
    if conf.selectedGC == gcUnselected:
```

**Review by zerbina — CHANGES_REQUESTED** [review]
I think handling of the "default" GC should stay in the options processing / front-end of the compiler.

Please use `gcUnselected` to implement the "default" option. It reduces the amount of "unreachable" states that the rest of the compiler has to consider.

**Pull request closed without merging**